### PR TITLE
CoffeeScript -> JS

### LINKED ISSUE #12: Peer dependency warning with Hubot 3
Installing this package together with Hubot 3 generates `warning " > hubot-youtube@1.0.2" has incorrect peer dependency "hubot@2.x".` I suspect that this package is compatible with Hubot 3, so a simple update of the peer dependency range should be sufficient?

### PR DESCRIPTION
**Pull request by TimvdLippe**
This PR updates all coffeescript files to JS ES6 to be compatible with Hubot 3.

Fixes #12

### CODE REVIEW

**Comment by stephenyeargin** [comment]
@TimvdLippe If you have a moment, tell me a bit more about the Hubot v3 requirements (namely, where I can keep up with them). I maintain several packages and am curious how much work is ahead. 😬

File: package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
   },
   "dependencies": {},
   "peerDependencies": {
-    "hubot": "2.x"
+    "hubot": "3.x"
```
Comment (stephenyeargin) [inline package.json]: Will this make it no longer viable for v2 Hubots?

File: index.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/*
+ * decaffeinate suggestions:
+ * DS101: Remove unnecessary use of Array.from
+ * DS102: Remove unnecessary code created because of implicit returns
+ * DS205: Consider reworking code to avoid use of IIFEs
+ * DS207: Consider shorter variations of null checks
+ * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
+ */
```
Comment (stephenyeargin) [inline index.js]: We can probably delete this block. As an aside, the new `hubot:script` generator probably has a better version of this?

**Comment by TimvdLippe** [comment]
@stephenyeargin Basically as far as I can understand, Hubot 3 is the same as Hubot 2, but all the coffeescript related stuff is removed. With these changes, I was able to make it run again. The other breaking changes are described at https://github.com/hubotio/hubot/releases/tag/v3.0.0

File: package.json (hunk 1 above)
Comment (TimvdLippe) [reply package.json]: Yes that is correct.

File: index.js (hunk 2 above)
Comment (TimvdLippe) [reply index.js]: Yup did this locally but forgot to do it here.

**Comment by stephenyeargin** [comment]
>Basically as far as I can understand, Hubot 3 is the same as Hubot 2, but all the coffeescript related stuff is removed.

Yeah, I noticed that in core. Wasn't sure what the plan was for the community packages if they _had_ to move to ES6 or if they were fine as-is. The `peerDependency` change probably made the bigger difference for your core issue, but I'm wondering if there's a notation we can use to indicate either `2.x` or `3.x`.

@hubot-scripts/hubot-youtube If you have a moment to weigh in on Hubot^Next as it relates to packages, that would be enlightening. :sparkles:

**Comment by TimvdLippe** [comment]
> but I'm wondering if there's a notation we can use to indicate either 2.x or 3.x.

There certainly is, but sadly Hubot 3 is incapable of consuming a package with just coffeescript files. This forces the package to convert to JS instead 😢

**Comment by stephenyeargin** [comment]
Weird. I feel like that'd be a bigger announcement somewhere than what's in the release notes. 🤔

File: index.js (hunk 2 above)
Comment (TimvdLippe) [reply index.js]: Done :+1:

**Comment by TimvdLippe** [comment]
@stephenyeargin Friendly ping on this PR 😄

**Comment by stephenyeargin** [comment]
I'd really hoped for a bit of feedback from the core Hubot folks to see if the CoffeeScript deprecation was permanent/required for future work. This isn't a terribly consequential package, so I'll go forward with merging.

**Comment by TimvdLippe** [comment]
@stephenyeargin Awesome, thanks for the quick response!

**Comment by stephenyeargin** [comment]
Released with `v1.10`.

**Pull request merged**